### PR TITLE
Abstract classes usage in python.

### PR DESCRIPTION
**Pull request by asenina**

### CODE REVIEW

**Review by vshampor — APPROVED** [review]

File: docs/styleguide/PyGuide.md [hunk 1]
```diff
@@ -236,6 +237,31 @@ with open("hello.txt") as hello_file:
         print(line)
 ```
 
+
+<a id="s2.9-abstract-classes"></a>
+<a id="29-abstract-classes"></a>
+<a id="abstract-classes"></a>
+### 2.9 Abstract Classes 
+
+When defining user abstract classes, the following template should be used:
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:246]:
```suggestion
When defining abstract classes, the following template should be used:
```

File: docs/styleguide/PyGuide.md [hunk 2]
```diff
@@ -236,6 +237,31 @@ with open("hello.txt") as hello_file:
...
+
+```python
+from abc import ABC, abstractmethod
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:249]: Are you allowing an abstract property? Could you add this to the template?

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: docs/styleguide/PyGuide.md (hunk 2 above)
Comment (asenina) [reply docs/styleguide/PyGuide.md:249]: Done.

File: docs/styleguide/PyGuide.md [hunk 3]
```diff
@@ -236,6 +237,38 @@ with open("hello.txt") as hello_file:
...
+        Calculate accuracy metric for model.
+        """
+
+    @property
+    @abstractmethod
+    def sample_generator(self) -> `SampleGenerator`:
+      """
+      :return: The instance of `SampleGenerator`.
+      """
+
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:268]: What about classmethod, statismethod, "internal" methods, property setter?

```suggestion
from abc import ABC, abstractmethod

class C(ABC):
    @abstractmethod
    def my_abstract_method(self, ...):
        ...
    @classmethod
    @abstractmethod
    def my_abstract_classmethod(cls, ...):
        ...
    @staticmethod
    @abstractmethod
    def my_abstract_staticmethod(...):
        ...
    @property
    @abstractmethod
    def my_abstract_property(self):
        ...
    @my_abstract_property.setter
    @abstractmethod
    def my_abstract_property(self, val):
        ...
    @abstractmethod
    def _get_x(self):
        ...
    @abstractmethod
    def _set_x(self, val):
```

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: docs/styleguide/PyGuide.md (hunk 3 above)
Comment (asenina) [reply docs/styleguide/PyGuide.md:268]: It looks good for me.

**Comment by asenina** [comment]
Jenkins please retry a build

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**